modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/client/AsahFaroBackendClientImpl.java

```diff
@@ -370,6 +352,61 @@ private MultivaluedMap<String, Object> _getParameters(
 		return uriVariables;
 	}
 
+	private String _getResponse(
+			long companyId, Http.Method method,
+			MultivaluedMap<String, Object> parameters, String path,
+			Object object)
+		throws Exception {
+
+		AnalyticsConfiguration analyticsConfiguration =
+			_analyticsSettingsManager.getAnalyticsConfiguration(companyId);
+
+		String url = StringBundler.concat(
+			analyticsConfiguration.liferayAnalyticsFaroBackendURL(),
+			StringPool.SLASH, path);
+
+		Http.Options options = new Http.Options();
+
+		if (parameters != null) {
+			for (MultivaluedMap.Entry<String, List<Object>> entry :
+					parameters.entrySet()) {
+
+				for (Object value : entry.getValue()) {
+					url = HttpComponentsUtil.addParameter(
+						url, entry.getKey(), value.toString());
+				}
+			}
+		}
+
+		if (object != null) {
```

Comment:
As you are calculating the URL at this point, this part of the code should be moved above the URL calculations

Reply:
ok, done!